presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -841,31 +843,36 @@ private List<Partition> batchGetPartition(String databaseName, String tableName,
     @Override
     public MetastoreOperationResult addPartitions(MetastoreContext metastoreContext, String databaseName, String tableName, List<PartitionWithStatistics> partitions)
     {
+        List<CompletableFuture<Future<BatchCreatePartitionResult>>> futures = new ArrayList<>();
```

Comment:
This can be just List<CompletableFuture<BatchCreatePartitionResult>>

Reply:
this is used to list all CompletableFuture to hold each async batch request for which each batch request is a future call.

Reply:
Type of this is what is returned in thenApply call(which is just null). So you can make this List<CompletableFuture<Void>>